SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriverGameplayDebuggerContext.cpp

```diff
@@ -242,6 +255,84 @@ void USpatialNetDriverGameplayDebuggerContext::TickServer()
 
 		It.RemoveCurrent();
 	}
+
+	for (auto& Pair : TrackedEntities)
+	{
+		if (Pair.Value.Component.TrackPlayer)
+		{
+			if (!NetDriver->HasServerAuthority(Pair.Key))
+			{
+				// Do not track player if we don't have authority
+				continue;
+			}
+
+			TWeakObjectPtr<AGameplayDebuggerCategoryReplicator> ReplicatorWeakObjectPtr = Pair.Value.ReplicatorWeakObjectPtr;
+			AGameplayDebuggerCategoryReplicator* Replicator = ReplicatorWeakObjectPtr.Get();
+			if (Replicator == nullptr)
+			{
+				// Not unexpected - assume latency and wait for actor to appear
+				continue;
+			}
+
+			const AActor* PlayerControllerActor = Cast<AActor>(Replicator->GetReplicationOwner());
+			if (!ensureMsgf(PlayerControllerActor != nullptr, TEXT("Serious error: Cannot find player controller for replicated gameplay debugger actor")))
+			{
+				continue;
+			}
+
+			check(NetDriver->LoadBalanceStrategy);
+			auto GetActorVirtualWorkerId = [](const AActor& InActor, const USpatialNetDriver& InNetDriver, const SpatialGDK::FSubView& InSubView)->VirtualWorkerId
+			{
+				check(InNetDriver.PackageMap);
+
+				const Worker_EntityId EntityId = InNetDriver.PackageMap->GetEntityIdFromObject(&InActor);
+				if (!ensureMsgf(EntityId != SpatialConstants::INVALID_ENTITY_ID, TEXT("Serious error: Cannot find actor in package map")))
+				{
+					return SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+				}
+
+				for (const SpatialGDK::ComponentData& Data : InSubView.GetView()[EntityId].Components)
+				{
+					if (Data.GetComponentId() == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID)
```

Comment:
I'm a bit surprised that the components on an entity in a view aren't indexed in a map? Is iterating over all of them really the only way to get to a specific one? This seems very odd to me, especially because the GDK is known to generate hundreds of components.

Reply:
It appeared that way to me, but I will double check.